src/function/scalar/list/list_resize.cpp

```diff
@@ -0,0 +1,253 @@
+		return;
+	}
+	D_ASSERT(result.GetType().id() == LogicalTypeId::LIST);
+	auto count = args.size();
+
+	result.SetVectorType(VectorType::FLAT_VECTOR);
+
+	auto &lists = args.data[0];
+	optional_ptr<Vector> child = &ListVector::GetEntry(args.data[0]);
+	auto &new_sizes = args.data[1];
+
+	UnifiedVectorFormat list_data;
+	lists.ToUnifiedFormat(count, list_data);
+	auto list_entries = reinterpret_cast<list_entry_t *>(list_data.data);
+
+	UnifiedVectorFormat new_size_data;
+	new_sizes.ToUnifiedFormat(count, new_size_data);
+	auto new_size_entries = reinterpret_cast<int64_t *>(new_size_data.data);
+
+	UnifiedVectorFormat child_data;
+	child->ToUnifiedFormat(count, child_data);
+	optional_ptr<T> child_entries;
+	if (child->GetType() != LogicalTypeId::LIST && child->GetType().InternalType() != PhysicalType::STRUCT) {
+		child_entries = (T *)child_data.data;
+	}
+
+	// Find the new size of the result child vector
+	idx_t new_child_size = 0;
+	for (idx_t i = 0; i < count; i++) {
+		auto index = new_size_data.sel->get_index(i);
+		if (new_size_data.validity.RowIsValid(index)) {
+			new_child_size += new_size_entries[index];
+		}
+	}
+
+	// Create the default vector if it exists
+	T *default_entries = nullptr;
+	UnifiedVectorFormat default_data;
+	optional_ptr<Vector> default_vector;
+	if (args.ColumnCount() == 3) {
+		default_vector = &args.data[2];
+		default_vector->ToUnifiedFormat(count, default_data);
+		default_entries = (T *)default_data.data;
+	}
+
+	ListVector::Reserve(result, new_child_size);
+	ListVector::SetListSize(result, new_child_size);
+
+	auto result_entries = FlatVector::GetData<list_entry_t>(result);
+	auto &result_child = ListVector::GetEntry(result);
+	auto result_child_data = FlatVector::GetData<T>(result_child);
+
+	// for each lists in the args
+	idx_t result_child_offset = 0;
+	for (idx_t args_index = 0; args_index < count; args_index++) {
+		auto l_index = list_data.sel->get_index(args_index);
+		auto new_index = new_size_data.sel->get_index(args_index);
+
+		// set null if lists is null
+		if (!list_data.validity.RowIsValid(l_index)) {
+			FlatVector::SetNull(result, args_index, true);
+			continue;
+		}
+
+		idx_t new_size_entry = 0;
+		if (new_size_data.validity.RowIsValid(new_index)) {
+			new_size_entry = new_size_entries[new_index];
+		}
+
+		// find the smallest size between lists and new_sizes
+		auto values_to_copy = MinValue<idx_t>(list_entries[l_index].length, new_size_entry);
+
+		// set the result entry
+		result_entries[args_index].offset = result_child_offset;
+		result_entries[args_index].length = new_size_entry;
+
+		// copy the values from the child vector
+		for (idx_t list_index = 0; list_index < values_to_copy; list_index++) {
+			if (!child_data.validity.RowIsValid(list_entries[l_index].offset + list_index)) {
+				FlatVector::SetNull(result_child, result_child_offset, true);
+			} else {
+				COPY_FUNCTION::template Operation<T>(result_child_data, child_entries, child, result_child,
```

Comment:
Could we use `VectorOperations::Copy` here instead of using templating to copy the data? I think that could simplify the code considerably particularly for nested types - plus it would likely be faster than using `GetValue` and `SetValue`.

For replicating the default value a constant vector could be created and pushed into the `VectorOperations::Copy`.